### PR TITLE
Allow send_h from open stream state

### PR DESCRIPTION
**Pull request by joedevivo**
Fixes 

```
12:15:10.317 [warning] Some unexpected message in open state. {send_h,[{<<":status">>,<<"400">>}]}, {stream_state,1,<0.399.0>,{ssl,{sslsocket,{gen_tcp,#Port<0.88376>,tls_connection,<0.376.0>},<0.397.0>}},idle,...
```

### CODE REVIEW

File: src/h2_stream.erl [hunk 1]
```diff
@@ -363,6 +363,14 @@ open({send_frame,
                 open
         end,
     {next_state, NextState, Stream};
+open(
+  {send_h, Headers},
+  #stream_state{}=Stream) ->
+    {next_state,
+     open,
```
Comment (stillinbeta) [inline src/h2_stream.erl]: in my case I'm sending an error message, so I want to send an `RST_STREAM`. But that's a seperate issue perhaps

File: src/h2_stream.erl [hunk 2]
```diff
@@ -114,10 +114,10 @@ start_link(StreamId, Connection, CallbackModule, Socket) ->
 send_pp(Pid, Headers) ->
     gen_fsm:send_event(Pid, {send_pp, Headers}).
 
--spec send_frame(pid(), http2_frame:frame()) ->
+-spec send_data(pid(), http2_frame:frame()) ->
```
Comment (stillinbeta) [inline src/h2_stream.erl]: why this change?

File: src/h2_stream.erl (hunk 1 above)
Comment (joedevivo) [inline src/h2_stream.erl]: It is separate. We can deal with that somewhere else if we need to.

File: src/h2_stream.erl (hunk 2 above)
Comment (joedevivo) [inline src/h2_stream.erl]: send_frame sounds more generic. This only works for data frames, and I changed recv_frame to recv_data a while ago.

Comment (stillinbeta) [inline src/h2_stream.erl]: could we change the type signature to reflect this?

Comment (joedevivo) [inline src/h2_stream.erl]: there isn't actually a data frame type. we could do `{frame_header(), h2_frame_data:payload()}` but I'm not sure it's worth it

Comment (stillinbeta) [inline src/h2_stream.erl]: maybe just a comment, then? "this can only send data frames"

Comment (joedevivo) [inline src/h2_stream.erl]: done. but that's not all. I'll do another PR later that introduces more explicit frame types

**Comment by stillinbeta** [comment]
:+1:
